File: Livro.c
#include <stdio.h>
#include <string.h>

typedef struct {
  char titulo[15];
  char autor[18];
  int anoPublicacao;
  int copiasDisponiveis;

} Livro;

void inicializar(Livro *c){
  strcpy(c->titulo, "Naruto");
  strcpy(c->autor, "Masashi Kishimoto");
  c->anoPublicacao = 1999;
  c->copiasDisponiveis = 1;
}

int existeCopia(Livro *c){
  if(c->copiasDisponiveis > 0){
    return 1;
  }return 0;
}

void emprestarCopia(Livro *c) {
  if(&existeCopia){
    c->copiasDisponiveis--;
    printf("Livro emprestado com sucesso!\n");
  }else{
    printf("Não há cópias disponíveis para empréstimo.\n");
  }
}
void devolverCopia(Livro *c) {
  c->copiasDisponiveis++;
  printf("Livro devolvido com sucesso!\n");
}

void informacoesLivro(Livro *c) {
  printf("Título: %s\n", c->titulo);
  printf("Autor: %s\n", c->autor);
  printf("Ano de Publicação: %i\n", c->anoPublicacao);
  printf("Cópias Disponíveis: %i\n", c->copiasDisponiveis);
}

int main(){
  Livro livro;
  inicializar(&livro);
  informacoesLivro(&livro);
}

File: Livro.java
package Java;

public class Livro {
    private String titulo = "Naruto";
    private String autor = "Masashi Kishimoto";
    private int anoPublicacao = 1999;
    private int copiasDisponiveis = 0;

    public void emprestarCopia() {
        if (existeCopia()) {
            System.out.println("Livro emprestado com sucesso!");
            this.copiasDisponiveis--;
        } else {
            System.out.println("Não há cópias desse livro.");
        }
    }

    public void devolverCopia() {
        System.out.println("Livro devolvido com sucesso!");
        this.copiasDisponiveis++;
    }

    public boolean existeCopia() {
        if (this.copiasDisponiveis > 0) {
            return true;
        }
        return false;
    }

    public void informacoesLivro() {
        System.out.println("Titulo: " + this.titulo);
        System.out.println("Autor: " + this.autor);
        System.out.println("Ano de Publicação: " + this.anoPublicacao);
        System.out.println("Cópias disponíveis: " + this.copiasDisponiveis);
    }

    public void main(String[] args) {
        informacoesLivro();
    }
}